### PR TITLE
Some new types

### PR DESCRIPTION
**Pull request by michielbdejong**
Here is a list of document types that Tosback2 uses but src/types.js does not yet list. We probably don't want to merge all of these, some include the service name in the document type and some are too similar or even look like typos. I'll tick the boxes of the ones this PR keeps, I'll update the Tosback2 rules at the XML source, not to use any document types that are not ticked here:

* [x] API Terms of Use
* [ ] APIs Terms of Use
* [x] Acceptable Use Policy
* [x] All Policies
* [x] Application-Based Services Terms of Use
* [ ] Cable Internet Terms of Use
* [ ] Canary Privacy Policy
* [ ] Cbs Interactive Privacy Policy
* [ ] Closed Captioning Policy
* [x] Conditions of Use
* [x] Consumer Terms of Sale
* [x] Cookie Policy
* [x] Copyright
* [ ] Copyright and Your use of the British Library Website
* [x] Customer Privacy Policy
* [x] DMCA
* [x] Data Policy
* [x] Data Use Policy
* [x] EULA
* [x] Etiquette Policy
* [ ] Flickr Privacy Policy
* [ ] GOOGLE PRIVACY POLICY
* [x] Geo guidelines
* [ ] Gizmo Privacy Policy
* [ ] Google Project Hosting - Additional Terms
* [ ] Group Video Calling Fair Usage Policy
* [ ] Host Guarantee Terms and Conditions
* [ ] Intellectual Property
* [ ] Intellectual Property Policy
* [ ] Legal Information
* [ ] Legal Information (Intuit)
* [ ] LinkedIn in Microsoft Applications with Your Personal Account
* [ ] Microsoft Services Agreement
* [ ] Microsoft Terms of Use
* [ ] Network User Agreement
* [ ] Oath Privacy Center
* [ ] Online Practices
* [ ] PRIVACY POLICY
* [ ] Policies and Notices
* [ ] Privacy
* [ ] Privacy Notice
* [ ] Privacy Policy Agreement
* [ ] Privacy Policy and Terms of Use
* [ ] Privacy and Cookies Policy
* [ ] Privacy for eero Devices
* [ ] Residential Services Subscriber Agreement
* [ ] Router Compatibility List
* [ ] Rules on Resolving Image Piracy Complaints
* [ ] SAMSUNG PRIVACY POLICY FOR THE U.S.
* [ ] Security
* [ ] Security & Privacy
* [ ] Security Advisory
* [ ] Signal Terms & Privacy Policy
* [ ] Term of Service
* [ ] Terms
* [ ] Terms & Conditions
* [ ] Terms Of Use
* [ ] Terms and Conditions and Privacy Policy
* [x] Terms and Conditions of Use
* [x] Terms of Sale
* [ ] Terms of Service & Privacy Policy
* [ ] Terms of Service 1
* [ ] Terms of Service and License Agreement
* [ ] Terms of Service and Privacy
* [ ] Terms of Use (Consumer)
* [ ] Terms of Use - About Copyright
* [ ] Terms of Use and Privacy Policy
* [ ] Third Party Advertising, Third Party Cookies, Affiliation, and Terms of Usage
* [x] ~~US~~ Acceptable Use Policy
* [ ] Universal Terms Of Service
* [x] User Agreement
* [ ] Visible Service Terms & Conditions
* [x] Visitor Agreement
* [x] Vunerability Disclosure Program
* [ ] Web Notices and Terms of Use
* [x] Website Terms of Service
* [ ] World Of Warcraft Terms Of Use Agreement
* [ ] YOUR PRIVACY & SECURITY
* [ ] end-user-license-agreement

### CODE REVIEW

**Comment by michielbdejong** [comment]
Not sure how to handle this one, it's basically two privacy policies that apparently both apply:
```xml
<docname name="Privacy Policy">
    <url name="https://policies.oath.com/us/en/oath/privacy/index.html" xpath="//main[@id=&apos;site-main&apos;]" reviewed="true">
      <norecurse name="arbitrary" />
    </url>
  </docname>
  <docname name="Flickr Privacy Policy">
    <url name="https://policies.oath.com/us/en/oath/privacy/products/flickr/index.html" xpath="//main[@id=&apos;site-main&apos;]" reviewed="true">
      <norecurse name="arbitrary"/>
    </url>
  </docname>
```

**Comment by michielbdejong** [comment]
I realized that I can't do the full tosback import until we have language support, so I'll put this PR up for review with the additions I made so far, and then we can follow up later once we have support for >=1 docs of the same type for the same service, which is what some of these need.

**Comment by michielbdejong** [comment]
I don't know all the implications of adding (too) many types that are similar to each other, but in general services use very different titles for their legal documents. Some call it Terms of Use, others call it Usage Policy. Should we allow this diversity or force a more strict small set of document types?

**Comment by michielbdejong** [comment]
This does greatly increase the number of times
```sh
(Tests skipped for this document type as it is not declared for this services)
```
is printed when you run `npm run validate`... :/

**Comment by MattiSG** [comment]
Thank you very much @michielbdejong!

> I'll use `privacy` for the product-specific one, and `privacyFull` for the parent company one.

We encountered this issue with Skype (parent company Microsoft). In #53, we introduced a document type `tosParent`. We then found that Skype had a consolidated ToS document which we imported, removing the document type along the way, in #76.
If there is no such consolidated document, then let's re-introduce `tosParent` from #53 🙂

> I can't do the full tosback import until we have language support

Just to double-check: do you mean being able to store documents in different languages? Is the issue the fact that some documents are not in English (which contradicts our current contribution guidelines), or is it that some ToSback documents are tracked in several languages?

> Should we allow this diversity or force a more strict small set of document types?

In order to enable comparative analysis, I would find it great to define _types_ independently from the provider-given name. This way, we could compare terms of services or privacy policies across companies, no matter how they call them in their own lingo. This seems reinforced by the future addition of multiple languages: I don't think it would be manageable to translate each type into several languages!

I understand this means being able to better define each type, so that one can find the most appropriate one beyond the names. I see a couple ways this could be solved:

1. Add a `description` field to the document types, which would enable a more complete description and would help contributors identify the most appropriate type no matter the provider-given name.
2. Add a `synonyms` field to the document types, which would help with disambiguating.
3. Characterise each document type along three dimensions of the contractual relationship: _party 1_ / _party 2_ /  _object_ . [Example 1](https://github.com/ambanum/CGUs/pull/77#discussion_r465201769), [example 2](https://github.com/ambanum/CGUs/pull/77#discussion_r465202680).

This would mean, for example, that:

- A “cookie policy” describes the agreement on _cookies_ between _service provider_ and _individual end user_.
- A “privacy policy” describes the agreement on _personal data_ between _service provider_ and _individual end user_.
- A “law enforcement guideline” describes the agreement on _service_ between _service provider_ and _law enforcement agents_.
- “commercial terms” describe the agreement on _service_ between _service provider_ and _commercial intermediaries_.

I am not yet sure that this characterization is a proper theoretical representation of a contractual relationship, but it does seem to help pragmatically in defining document types. What do you think of it @michielbdejong? Do you have recommendations of people to reach out to to discuss this topic? 🙂 

Twitter is really the worst for that (or the best if one considers readability!), as their [rules and policies](https://help.twitter.com/en/rules-and-policies) are split in dozens of documents. We added a few in #85, but there are definitely many others.

> This does greatly increase the number of times […] is printed when you run `npm run validate`... :/

I think this workaround is showing its limits. We'll probably have to find a better one.

File: src/types.js [hunk 1]
```diff
@@ -1,12 +1,72 @@
 export const TYPES = {
   tos: {
-    name: 'Terms of service',
+    name: 'Terms of Service',
     fileName: 'terms_of_service'
   },
+  website_tos: {
```
Comment (MattiSG) [inline src/types.js:6]: In favor of merging with `tos`, unless I misunderstand and there is a difference.

File: src/types.js [hunk 2]
```diff
@@ -1,12 +1,72 @@
 export const TYPES = {
   tos: {
```
Comment (MattiSG) [inline src/types.js:2]: Maybe we should expand acronyms 🤔

File: src/types.js [hunk 3]
```diff
@@ -1,12 +1,72 @@
...
+  },
+  visitor_agreement: {
+    name: 'Visitor Agreement',
+    fileName: 'visitor_agreement'
+  },
+  vendor_tos: {
+    name: 'Vendor Terms of Service',
+    fileName: 'vendor_terms_of_service'
+  },
+  tou: {
```
Comment (MattiSG) [inline src/types.js:22]: In favor of merging with `tos`, unless I misunderstand and there is a difference.

File: src/types.js [hunk 4]
```diff
@@ -1,12 +1,72 @@
...
+    name: 'Terms of Use',
+    fileName: 'terms_of_use'
+  },
+  acceptableUsePolicy: {
+    name: 'Acceptable Use Policy',
+    fileName: 'acceptable_use_policy'
+  },
+  termsAndConditions: {
```
Comment (MattiSG) [inline src/types.js:30]: In favor of merging with `tos`, unless I misunderstand and there is a difference.

File: src/types.js [hunk 5]
```diff
@@ -1,12 +1,72 @@
...
+    name: 'Terms and Conditions',
+    fileName: 'terms_and_conditions'
+  },
+  termsAndPrivacyPolicy: {
```
Comment (MattiSG) [inline src/types.js:34]: We can add this, or we can ask contributors to select the subset of the document which is about ToS and about privacy, especially now that it's been made easier with #70… Could we have some examples of this @michielbdejong? 🙂

File: src/types.js [hunk 6]
```diff
@@ -1,12 +1,72 @@
...
+    name: 'Terms and Privacy Policy',
+    fileName: 'terms_and_privacy_policy'
+  },
+  termsOfUseAndConditionsOfUse: {
```
Comment (MattiSG) [inline src/types.js:38]: In favor of merging with `tos`, unless I misunderstand and there is a difference.

File: src/types.js [hunk 7]
```diff
@@ -1,12 +1,72 @@
...
+    name: 'Terms of Use and Conditions of Use',
+    fileName: 'terms_of_use_and_conditions_of_use'
+  },
+  conditionsOfUse: {
```
Comment (MattiSG) [inline src/types.js:42]: In favor of merging with `tos`, unless I misunderstand and there is a difference.

File: src/types.js [hunk 8]
```diff
@@ -1,12 +1,72 @@
...
+  },
+  termsOfSale: {
+    name: 'Terms of Sale',
+    fileName: 'terms_of_sale'
+  },
   privacy: {
     name: 'Privacy Policy',
     fileName: 'privacy_policy'
   },
+  privacyFull: {
```
Comment (MattiSG) [inline src/types.js:54]: In favor of merging with `privacy`, unless I misunderstand and there is a difference.

File: src/types.js [hunk 9]
```diff
@@ -1,12 +1,72 @@
...
+    name: 'Full Privacy Policy',
+    fileName: 'full_privacy_policy'
+  },
+  dataPolicy: {
```
Comment (MattiSG) [inline src/types.js:58]: In favor of merging with `privacy`, unless I misunderstand and there is a difference.
Could we have an example?

File: src/types.js [hunk 10]
```diff
@@ -1,12 +1,72 @@
...
+    name: 'Data Policy',
+    fileName: 'data_policy'
+  },
+  dataUsePolicy: {
```
Comment (MattiSG) [inline src/types.js:62]: In favor of merging with `privacy` or `data`, unless I misunderstand and there is a difference.

File: src/types.js [hunk 11]
```diff
@@ -1,12 +1,72 @@
...
+    name: 'Data Use Policy',
+    fileName: 'data_use_policy'
+  },
+  privacyStatement: {
```
Comment (MattiSG) [inline src/types.js:66]: In favor of merging with `privacy`, unless I misunderstand and there is a difference.

File: src/types.js [hunk 12]
```diff
@@ -19,18 +79,46 @@ export const TYPES = {
     name: 'Community Guidelines',
     fileName: 'community_guidelines'
   },
+  customerPrivacyPolicy: {
+    name: 'Costomer Privacy Policy',
```
Comment (MattiSG) [inline src/types.js:83]:
```suggestion
    name: 'Customer Privacy Policy',
```

File: src/types.js [hunk 13]
```diff
@@ -19,18 +79,46 @@ export const TYPES = {
...
+    fileName: 'costumer_privacy_policy'
```
Comment (MattiSG) [inline src/types.js:84]:
```suggestion
    fileName: 'customer_privacy_policy'
```

File: src/types.js [hunk 14]
```diff
@@ -19,18 +79,46 @@ export const TYPES = {
     name: 'Community Guidelines',
     fileName: 'community_guidelines'
   },
+  customerPrivacyPolicy: {
```
Comment (MattiSG) [inline src/types.js:82]: In favor of merging with `privacy`, unless I misunderstand and there is a difference.

File: src/types.js [hunk 15]
```diff
@@ -19,18 +79,46 @@ export const TYPES = {
...
+    name: 'Costomer Privacy Policy',
+    fileName: 'costumer_privacy_policy'
+  },
+  DMCA: {
+    name: 'DMCA',
+    fileName: 'dmca'
+  },
+  EULA: {
```
Comment (MattiSG) [inline src/types.js:90]: In favor of merging with `tos`, unless I misunderstand and there is a difference.

File: src/types.js [hunk 16]
```diff
@@ -19,18 +79,46 @@ export const TYPES = {
...
+    name: 'EULA',
+    fileName: 'eula'
+  },
   brandGuidelines: {
     name: 'Brand Guidelines',
     fileName: 'brand_guidelines'
   },
+  geoGuidelines: {
```
Comment (MattiSG) [inline src/types.js:98]: I don't understand what this is. Could we have an example?

File: src/types.js [hunk 17]
```diff
@@ -19,18 +79,46 @@ export const TYPES = {
...
+  },
   reviewGuidelines: {
     name: 'Review Guidelines',
     fileName: 'review_guidelines'
   },
+  cookiePolicy: {
+    name: 'Cookie Policy',
+    fileName: 'cookie_policy'
+  },
+  cookieAndAdPolicy: {
```
Comment (MattiSG) [inline src/types.js:110]: In favor of merging with `cookiesPolicy`, unless I misunderstand and there is a difference.

File: src/types.js [hunk 18]
```diff
@@ -19,18 +79,46 @@ export const TYPES = {
...
   },
+  geoGuidelines: {
+    name: 'Geo Guidelines',
+    fileName: 'geo_guidelines'
+  },
   reviewGuidelines: {
     name: 'Review Guidelines',
     fileName: 'review_guidelines'
   },
+  cookiePolicy: {
```
Comment (MattiSG) [inline src/types.js:106]: In favor of merging with `cookiesPolicy`.

File: src/types.js [hunk 19]
```diff
@@ -19,18 +79,46 @@ export const TYPES = {
...
+  },
+  cookieAndAdPolicy: {
+    name: 'Cookie and Ad Policy',
+    fileName: 'cookie_and_ad_policy'
+  },
   cookiesPolicy: {
     name: 'Cookies Policy',
     fileName: 'cookies_policy'
   },
+  copyright: {
```
Comment (MattiSG) [inline src/types.js:118]: Could we have examples, in order to understand if this is about describing copyright owners, about DMCA, or about providing copyright licenses?

File: src/types.js [hunk 20]
```diff
@@ -42,5 +130,17 @@ export const TYPES = {
   lawEnforcementGuidelines: {
     name: 'Law Enforcement Guidelines',
     fileName: 'law_enforcement_guidelines'
-  }
+  },
+  consumerTermsOfSale: {
+    name: 'Consumer Terms of Sale',
+    fileName: 'consumer_terms_of_sale'
+  },
+  securityAndPrivacy: {
```
Comment (MattiSG) [inline src/types.js:138]: In favor of merging with `privacy`, unless I misunderstand and there is a difference.
Could we have an example?

File: src/types.js [hunk 21]
```diff
@@ -42,5 +130,17 @@ export const TYPES = {
...
+    name: 'Security and Privacy',
+    fileName: 'security_and_privacy'
+  },
+  vulnerabilityDisclosureProgram: {
```
Comment (MattiSG) [inline src/types.js:142]:
```suggestion
  vulnerabilityDisclosurePolicy: {
```

**Comment by MattiSG** [comment]
> I'll update the Tosback2 rules at the XML source, not to use any document types that are not ticked here

I see that many of these types contain company names, and are thus trivial to realign (e.g. _SAMSUNG PRIVACY POLICY FOR THE U.S._ should be `privacy`).
Some others are syntactic variants, also trivial to realign (e.g. _APIs Terms of Use_ and _API Terms of Use_).

However some seem more complex, as they are specific to a service type, such as _Host Guarantee Terms and Conditions_. When these document types are not ticked, what is your intention with them? 🙂

**Comment by MattiSG** [comment]
> they are specific to a service type […] what is your intention with them?

Just saw that this was precisely your question in #89 😅

File: src/types.js (hunk 1 above)
Comment (michielbdejong) [reply src/types.js:6]: It's used by for instance Apple to define the ToS of the www.apple.com website, as opposed to their iChat ToS, Game Center ToS, iCloud ToS, and iTunes ToS. Maybe the right thing to do here is to define "the Apple website", iChat, Apple Game Center, iCloud and iTunes all as separate services. It's probably inaccurate to define "Apple" as a single service.

Comment (michielbdejong) [reply src/types.js:6]: That way, we can have "the `tos` of www.apple.com" instead of "the `website_tos` of Apple".

File: src/types.js (hunk 5 above)
Comment (michielbdejong) [reply src/types.js:34]: Good point, and it applies in a few other cases as well. I'll do that.

File: src/types.js (hunk 1 above)
Comment (MattiSG) [reply src/types.js:6]: > Maybe the right thing to do here is to define "the Apple website", iChat, Apple Game Center, iCloud and iTunes all as separate services.

Yes, strong +1.
This is also what I suggested and we finally decided in #77: Facebook, Facebook Ads and Facebook Payments should all be separate services. This suggests there might be a use for modeling the service _provider_ in the end, but right now let's declare each _service_ independently 🙂

File: src/types.js (hunk 2 above)
Comment (Ndpnt) [reply src/types.js:2]: Yep 👍 , I was thinking the same the last week.

**Comment by MattiSG** [comment]
Will be impacted by #90, inputs welcome there!

**Comment by michielbdejong** [comment]
Superseded by #90.

**Comment by MattiSG** [comment]
Has been only partly superseded by #90: some types are still missing.

@ndpnt and myself spent some time yesterday building a mapping table for all the document types you referenced in this PR 🙂

Since you mentioned wanting to edit the original XML files (we could also build a dynamic mapping table), this [mapping table](https://docs.google.com/spreadsheets/d/1RwnWbRdAyBGxOueK1yTv7li_KHmgNbszC4HrY9Lxx68/) maps each ToSBack type to a CGUs type.

Types in bold are new ones that should be created. In italic are documents that we believe should be dropped as the links are outdated.

**Pull request closed without merging**